### DISCUSSION TITLE
Apple Calendar Event Synching Issue on iPhone

### DISCUSSION THREAD

**Discussion by rohail-office** [spatie/icalendar-generator#50] [Bugs]
0

I am using the package spatie/icalendar-generator with Laravel to add event in Apple calendar

Version Details:
Spatie Verion = 1.5
Laravel Version = 7.24

By Using this package I am able create .ics file for adding event in calendar like Google, Outlook and Apple. On Google and Outlook the .ics file is working fine and adding the event in external calendar.

On MacBook 15 .ics file is working properly and synching the event in calendar but In iPhone 7+ ( 14.4.2) my .ics file not working properly and not adding event in iPhone calendar. It just showing the event details when we click on the .ics file.

**Comment by rubenvanassche** [comment]
hi @rohail-office,

Could you provide some code that is creating this calendar? So I can check out what's going wrong.

**Reply by solomoreal** [reply]
I have the same issue as well: protected function generateCalendar($employee)
    {
        $schedules = $employee->schedulers()->where('published', false)->whereDate('start_at', '>', now())->get();

        $calendar = Calendar::create('Employee Schedule');

        foreach ($schedules as $schedule) {
            $event = Event::create()
                ->name('work shift '.$schedule->company->name ?? 'Scheduled Event')
                ->description($schedule->shift_note ?? 'No description provided')
                ->uniqueIdentifier($schedule->id)
                ->createdAt(Carbon::parse($schedule->start_at))
                ->startsAt(Carbon::parse($schedule->start_at))
                ->endsAt(Carbon::parse($schedule->end_at))
                ->address($schedule->address ?? 'No location provided')
                ->coordinates($schedule->latitude, $schedule->longitude)
                ->alertMinutesBefore(10, 'Your shift will start in 10 minutes');

            $calendar->event($event);
        }

        return $calendar;
    }

**Reply by rubenvanassche** [reply]
So I tried the following:

```php
test('iphone stuff', function (){
    $calendar = Calendar::create('Employee Schedule');

    $schedules = [
        (object) [
            'id' => 1,
            'company' => (object) ['name' => 'Company A'],
            'shift_note' => 'Work shift from 9am to 5pm',
            'start_at' => '2021-08-01 09:00:00',
            'end_at' => '2021-08-01 17:00:00',
            'address' => '123 Main St, Springfield, IL',
            'latitude' => 39.781721,
            'longitude' => -89.650148,
        ],
        (object) [
            'id' => 2,
            'company' => (object) ['name' => 'Company B'],
            'shift_note' => 'Work shift from 10am to 6pm',
            'start_at' => '2021-08-02 10:00:00',
            'end_at' => '2021-08-02 18:00:00',
            'address' => '456 Elm St, Springfield, IL',
            'latitude' => 39.781721,
            'longitude' => -89.650148,
        ],
    ];

    foreach ($schedules as $schedule) {
        $event = Event::create()
            ->name('work shift '.$schedule->company->name ?? 'Scheduled Event')
            ->description($schedule->shift_note ?? 'No description provided')
            ->uniqueIdentifier($schedule->id)
            ->createdAt(Carbon::parse($schedule->start_at))
            ->startsAt(Carbon::parse($schedule->start_at))
            ->endsAt(Carbon::parse($schedule->end_at))
            ->address($schedule->address ?? 'No location provided')
            ->coordinates($schedule->latitude, $schedule->longitude)
            ->alertMinutesBefore(10, 'Your shift will start in 10 minutes');

        $calendar->event($event);
    }

    file_put_contents('calendar.ics', $calendar->get());
});
```

Could you try breaking this?

![IMG_0518](https://github.com/user-attachments/assets/00926eae-8009-409c-b978-886b7fa12cdd)
